modules/hal_gigadevice/CMakeLists.txt

```diff
@@ -6,20 +6,30 @@ if(CONFIG_HAS_GD32_HAL)
 string(TOUPPER ${CONFIG_SOC} gd32_soc_uc)
 
 set(gd32_soc_dir     ${ZEPHYR_HAL_GIGADEVICE_MODULE_DIR}/${CONFIG_SOC_SERIES})
-set(gd32_cmsis_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+if(CONFIG_SOC_FAMILY_GD32_ARM)
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+else()
+  set(gd32_soc_sys_dir   ${gd32_soc_dir}/riscv)
+  zephyr_include_directories(${gd32_soc_dir}/riscv/drivers)
+endif()
+
 set(gd32_std_dir     ${gd32_soc_dir}/standard_peripheral)
 set(gd32_std_src_dir ${gd32_std_dir}/source)
 
 zephyr_library_named(hal_gigadevice)
 
 zephyr_compile_definitions(${gd32_soc_uc})
 
+if(NOT "0" STREQUAL "${CONFIG_GD32_HXTAL_VALUE}")
+  zephyr_compile_definitions(HXTAL_VALUE=${CONFIG_GD32_HXTAL_VALUE})
```

Comment:
change it to zephyr_library_compile_definitions and define HXTAL_VALUE in soc/riscv/riscv-privilege/gd32vf103/soc.h.
And add HXTAL_VALUE define in soc.h.

The soc.h is including gd32vf103.h and the gd32vf103.h requires the definition of HXTAL_VALUE.
soc.h abstracting the gigadevice header file structure different among each SoC model.
GD32's peripheral driver uses that file instead of including headers directly from the external module directory.
Thus, HXTAL_VALUE needs to define even not under the hal_gigadevice directory.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/34970#discussion_r768755127